tmc-langs-qmake/src/main/java/fi/helsinki/cs/tmc/langs/qmake/QmakePlugin.java

```diff
@@ -121,13 +127,27 @@ protected StudentFilePolicy getStudentFilePolicy(Path projectPath) {
 
     @Override
     public RunResult runTests(Path path) {
-        Path shadowDir = makeShadowBuildDir(path);
+        Path fullPath;
+        try {
+            fullPath = path.toRealPath(LinkOption.NOFOLLOW_LINKS);
+        } catch (IOException e) {
+            log.error("Path does not exist", path);
+            throw new RuntimeException("Path does not exist", e);
+        }
+
+        Path shadowDir;
+        try {
+            shadowDir = makeShadowBuildDir(fullPath);
+        } catch (IOException e) {
+            log.error("Unable to make shadow build dir to path", fullPath);
+            throw new RuntimeException(e);
```

Comment:
Don't throw `RuntimeException`s since  they are unchecked exceptions in Java so there is no way for the consumer of this interface to expect this.